sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -1170,4 +887,344 @@ private void waitForFuturesToFinish() throws ExecutionException, InterruptedExce
       }
     }
   }
+
+  /**
+   * A {@link PTransform} to read data from Apache Cassandra. See {@link CassandraIO} for more
+   * information on usage and configuration.
+   */
+  @AutoValue
+  public abstract static class ReadAll<T> extends PTransform<PCollection<Read<T>>, PCollection<T>> {
+
+    @Nullable
+    abstract ValueProvider<List<String>> hosts();
```

Comment:
This transform should not have any attribute / with method, those come from the Read in the input PCollection.

Reply:
For ReadAll to be useable we need to establish connection information in the `setup` phase of the QueryFn, so we'll need all this before a `Read<T>` comes in.   I'd also like to keep the API between ReadAll and Read similar so it's easy for the end user to switch a `Read<T>` with a `ReadAll<T>`, and just operating on a different type.